client/chat/connected-channel-name.tsx

```diff
@@ -18,22 +27,57 @@ export interface ConnectedChannelNameProps {
 }
 
 /**
- * A component which, given a channel ID, displays a channel name. Pretty barebones at the moment,
- * but will be extended with further functionality soon.
+ * A component which, given a channel ID, displays a clickable channel name, which displays channel
+ * info when clicked.
  */
 export function ConnectedChannelName({ className, channelId }: ConnectedChannelNameProps) {
   const dispatch = useAppDispatch()
-  const channelInfo = useAppSelector(s => s.chat.idToInfo.get(channelId))
+  const basicChannelInfo = useAppSelector(s => s.chat.idToBasicInfo.get(channelId))
+  const isChannelDeleted = useAppSelector(s => s.chat.deletedChannels.has(channelId))
 
   useEffect(() => {
-    dispatch(getBatchChannelInfo(channelId))
-  }, [dispatch, channelId])
+    if (!isChannelDeleted) {
+      dispatch(getBatchChannelInfo(channelId))
+    }
+  }, [dispatch, channelId, isChannelDeleted])
 
-  const channelName = channelInfo?.name ?? (
-    <LoadingName aria-label={'Channel loading…'}>
-      &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
-    </LoadingName>
-  )
+  const [channelInfoCardOpen, openChannelInfoCard, closeChannelInfoCard] = usePopoverController()
+  const [anchor, anchorX, anchorY] = useAnchorPosition('right', 'top')
+
+  const onClick = useStableCallback((e: React.MouseEvent) => {
```

Comment:
I don't think either of these functions are necessary, you can just pass `openChannelInfoCard` and `closeChannelInfoCard` directly?